File: Go/arrays/arrays.go
package main

import (
	"fmt"
)

type Array struct {
	data     []int // Slice to hold the array elements
	size     int   // Current number of elements
	capacity int   // Maximum capacity of the array
}

// Initialize a new array with a given capacity
func NewArray(capacity int) *Array {
	return &Array{
		data:     make([]int, capacity),
		size:     0,
		capacity: capacity,
	}
}

// Insert an element into the array
func (arr *Array) Insert(element int) {
	if arr.size >= arr.capacity {
		fmt.Println("Array is full! Cannot insert element.")
		return
	}
	arr.data[arr.size] = element
	arr.size++
}

// Delete an element at a specific index
func (arr *Array) Delete(index int) {
	if index < 0 || index >= arr.size {
		fmt.Println("Invalid index!")
		return
	}
	for i := index; i < arr.size-1; i++ {
		arr.data[i] = arr.data[i+1]
	}
	arr.data[arr.size-1] = 0 // Clear the last element
	arr.size--
}

// Display the array contents
func (arr *Array) Display() {
	if arr.size == 0 {
		fmt.Println("Array is empty!")
		return
	}
	fmt.Print("Array elements: ")
	for i := 0; i < arr.size; i++ {
		fmt.Print(arr.data[i], " ")
	}
	fmt.Println()
}

func main() {
	// Initialize an array with a capacity of 10
	arr := NewArray(10)

	// Insert elements into the array
	arr.Insert(5)
	arr.Insert(10)
	arr.Insert(15)

	// Display the array contents
	arr.Display()

	// Delete an element at index 1
	arr.Delete(1)

	// Display the array contents again
	arr.Display()
}

File: Python/arrays/arrays.py
class Array:
    def __init__(self, capacity):
        self.data = [None] * capacity  # Initialize the array with None values
        self.size = 0  # Current number of elements
        self.capacity = capacity  # Maximum capacity of the array

    def insert(self, element):
        if self.size >= self.capacity:
            print("Array is full! Cannot insert element.")
            return
        self.data[self.size] = element
        self.size += 1

    def delete(self, index):
        if index < 0 or index >= self.size:
            print("Invalid index!")
            return
        for i in range(index, self.size - 1):
            self.data[i] = self.data[i + 1]
        self.data[self.size - 1] = None  # Clear the last element
        self.size -= 1

    def display(self):
        if self.size == 0:
            print("Array is empty!")
            return
        print("Array elements:", self.data[:self.size])

# Example usage
if __name__ == "__main__":
    # Initialize an array with a capacity of 10
    arr = Array(10)

    # Insert elements into the array
    arr.insert(5)
    arr.insert(10)
    arr.insert(15)

    # Display the array contents
    arr.display()

    # Delete an element at index 1
    arr.delete(1)

    # Display the array contents again
    arr.display()
